Tasks/PullRequestCodeAnalysis/PRCA/PrcaOrchestrator.ts

```diff
@@ -94,25 +95,40 @@ export class PrcaOrchestrator {
     public postSonarQubeIssuesToPullRequest(sqReportPath: string): Promise<void> {
         this.logger.LogDebug(`SonarQube report path: ${sqReportPath}`);
         if (sqReportPath === undefined || sqReportPath === null) {
-            throw new Error("Make sure a SonarQube enabled build task ran before this step.");
+            return Promise.reject('Make sure a SonarQube-enabled build task ran before this step.');
         }
 
         var allMessages:Message[] = this.sqReportProcessor.FetchCommentsFromReport(sqReportPath);
         var messagesToPost:Message[] = null;
-        return this.PrcaService.getModifiedFilesInPr()
+        return Promise.resolve()
+            .then(() => {
+                return this.PrcaService.getModifiedFilesInPr()
+                    .catch((error) => {
```

Comment:
This embedded catch serves to catch rejections that originate from _getModifiedFilesInPr()_ only, much like a conventional catch statement. We can then give a specific error message without the possibility of erroneously capturing rejections from further upstream.

Any rejection that does pass down from upstream will fall out of this Orchestrator (by design) and be caught by the master catch at the end of the PRCA task.